tiledb/common/src/query/condition/mod.rs

```diff
@@ -0,0 +1,930 @@
+#[cfg(feature = "proptest-strategies")]
```

Comment:
All the struct/enum definitions were moved from the `tiledb_api` crate so that I could use them in `test-utils/cells`.

There is some new stuff here, namely some of the `Not` impls and the `pub fn negate`